src/k4a_ros_bridge_node.cpp

```diff
@@ -44,11 +44,19 @@ int main(int argc, char** argv)
 
   if (result == K4A_RESULT_SUCCEEDED)
   {
-    rclcpp::spin(node);
+    rclcpp::Rate rate(0.5);
+    while (rclcpp::ok()) {
+
+      device->update();
 
-    RCLCPP_INFO(node->get_logger(),"ROS Exit Started");
+      rclcpp::spin_some(node);
+      rate.sleep();
```

Comment:
hmm this might affect the whole functioning of the node. Will it be able to publish all the frames in the time? Maybe sth else breaks?

Reply:
ok then I can call the update function in a seperate thread using the same approach and in that way the node spin won't be affected